### PR TITLE
[FIX] Console hide/show on enter pressing

### PR DESCRIPTION
**Pull request by themaoci**
this fixes a trouble with console staying open after enabling WSAD walking
- rewritted console.lua and otui

### CODE REVIEW

**Comment by themaoci** [comment]
im waiting if you agree on this merge :)

File: modules/game_console/console.lua [hunk 1]
```diff
@@ -177,25 +209,11 @@ function selectAll(consoleBuffer)
...
-end
-
-function updateChatMode()
-  if consoleToggleChat:isChecked() then
-    disableChat()
-  else
-    enableChat()
-  end
+  useAutoHideChat = consoleToggleChat:isChecked()
 end
```
Comment (oarsc) [inline modules/game_console/console.lua:213]: By modifying this method like this, you are breaking the hotkey configuration to change between wasd and chat mode in

https://github.com/mehah/otclient/blob/b3044107ce1991a736068dc24b6f847c57a284d7/modules/game_hotkeys/hotkeys_manager.lua#L442-L443

File: modules/game_console/console.lua [hunk 2]
```diff
@@ -374,6 +395,32 @@ function clear()
...
+  if not floatingMode then
+    return
+  end
+  -- update margin
+  return true
+end
+
+function onDragLeave(widget, pos)
+  return floatingMode
+end
```
Comment (oarsc) [inline modules/game_console/console.lua:423]: What does this functions do depending on the returned value? (I've no idea)

File: modules/game_console/console.otui [hunk 3]
```diff
@@ -66,7 +66,7 @@ Panel
     anchors.top: parent.top
     margin-left: 13
     margin-top: 8
-    @onCheckChange: updateChatMode()
+    @onCheckChange: toggleChat()
```
Comment (oarsc) [inline modules/game_console/console.otui:69]: Take into account the meaning of the functions name...
In your case, toggleChat doesn't really toggle anything.
This function is called once chat is toggled.

**Review by oarsc — CHANGES_REQUESTED** [review]
There are some changes that are not related with #147.
I noted some things that I found/didn't understand.

I have some doubts with your problem. You can set a hotkey to enable/disable chat (It can't be "enter", as enter is used to send messages and can't be used to disable chat).

Are you willing to do a behavior similar to Tibia? Using a temporal disable of WASD mode to send only 1 message?
If that is the case, it's not reacting to the checkbox being changed, it only reacts to 'enter'

https://user-images.githubusercontent.com/5507777/132962883-d62d6a2b-d03b-4367-978f-d073ac13e16e.mp4

File: modules/game_console/console.lua (hunk 2 above)
Comment (themaoci) [reply modules/game_console/console.lua:423]: yea i tried to add staff from otcv8 but i ended up abadoning it...

File: modules/game_console/console.lua (hunk 1 above)
Comment (themaoci) [reply modules/game_console/console.lua:213]: i will update it more and propose changes again i still didnt finished it yet cause there are no initial states unfortunatly...

**Comment by themaoci** [comment]
i will edit it abit more and ask for pull again then :)

**Pull request closed without merging**